### PR TITLE
hud action documentatino

### PR DESCRIPTION
**Pull request by simcha-waintrob**

### CODE REVIEW

File: actions/hud.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+
+
+### Return Object:
+The client will return this object
+
+#### ```Success```
+Indicate if the action is succedded.
+#### ```HUDKey```
+The key of the HUD.
```
Comment (lihi-sheffer) [inline actions/hud.md:66]: the hud key which is generated by the client

File: actions/hud.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+#### ```Message```
+The message that will be seen on the HUD (optional). 
+#### ```CancelMessage```
+The Cancel Message that will be seen on the HUD's cancel button (optional). 
+#### ```CancelEventKey```
+The event key that the client will emit to after the HUD's cancel button was pressed (mandatory if ```CancelMessage``` is provided). 
+#### ```Interval```
+Tell the client to poll after the interval time in seconds  (mandatory in 'show' state). 
+#### ```Key```
+The client's HUD key that the action will refer to (mandatory in both states 'show' and 'polling').
```
Comment (lihi-sheffer) [inline actions/hud.md:56]: the client should generate this key as response to show request. further requests will use this key

File: actions/hud.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        "Key": "8A6D7F6D-08AD-4386-85CF-35F726F73339"
+    }
+}
+```
+
+### Response
+```json
+{
+    "Success": true,
+    "HUDKey": "8A6D7F6D-08AD-4386-85CF-35F726F73339"
```
Comment (lihi-sheffer) [inline actions/hud.md:110]: the HudKey should be returned only from the show request.

File: actions/hud.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            updateMessage(`In progress ${i}%`);
+        }
+    },
+    // block of code that will run after the user pressed on the HUD'S cancel button.
+    cancelBlock: async () => {
+        console.log('cancel button clicked');
+    },
+};
+
+await pepperi.client.showHUD(hudOptions);
```
Comment (lihi-sheffer) [inline actions/hud.md:149]: consider return value - whether the block completed or canceled

Comment (simcha-waintrob) [reply actions/hud.md:149]: like this?
`{ 
        "cancled": true
}`

Comment (simcha-waintrob) [reply actions/hud.md:149]: @Chasky-H  what do you think?

File: actions/hud.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
+# Show HUD 
+Showing a progress HUD (Heads Up Display).
+
+## Description
+ It's helpful to give the user an indication that something happens now,
```
Comment (Chasky-H) [inline actions/hud.md:5]: something is happening

File: actions/hud.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+performing web requests or computing any computations that take a long time. \
```
Comment (Chasky-H) [inline actions/hud.md:6]: takes

File: actions/hud.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+The HUD can show a ```Message``` on the indicator.\
+In addition, the HUD can show a button that allow the user to close the HUD. the button will be shown only if ```CancelMessage``` text is provided. 
+
+the action have 3 states.
```
Comment (Chasky-H) [inline actions/hud.md:10]: The action has

File: actions/hud.md [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+                        user's code
+```
+
+client action
+
+
+### Parameters:
+
+#### ```State```
+The state of the HUD action (mandatory).
```
Comment (Chasky-H) [inline actions/hud.md:46]: The values of state can be:

(enum should be PascalCase)

File: actions/hud.md [hunk 9]
```diff
@@ -0,0 +1,156 @@
+# Show HUD 
+Showing a progress HUD (Heads Up Display).
+
+## Description
+ It's helpful to give the user an indication that something happens now,
+performing web requests or computing any computations that take a long time. \
+The HUD can show a ```Message``` on the indicator.\
+In addition, the HUD can show a button that allow the user to close the HUD. the button will be shown only if ```CancelMessage``` text is provided.
```
Comment (Chasky-H) [inline actions/hud.md:8]: What does the client do when the user presses the CancelMessage

File: actions/hud.md [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+#### ```State```
+The state of the HUD action (mandatory). 
+#### ```Message```
+The message that will be seen on the HUD (optional). 
+#### ```CancelMessage```
+The Cancel Message that will be seen on the HUD's cancel button (optional). 
+#### ```CancelEventKey```
+The event key that the client will emit to after the HUD's cancel button was pressed (mandatory if ```CancelMessage``` is provided). 
+#### ```Interval```
+Tell the client to poll after the interval time in seconds  (mandatory in 'show' state).
```
Comment (Chasky-H) [inline actions/hud.md:54]: The client should poll the emit-event loop after interval time (in seconds).
The client should do so using the CallbackKey of this action.

File: actions/hud.md [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+Tell the client to poll after the interval time in seconds  (mandatory in 'show' state). 
+#### ```Key```
+The client should generate this key as a response to show request. further requests on this hud instance ('polling' or 'hide') will use this key.
+
+
+### Return Object:
+The client will return this object
+
+#### ```Success```
+Indicate if the action is succedded.
```
Comment (Chasky-H) [inline actions/hud.md:63]: if the action succeeded

File: actions/hud.md [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+#### ```HUDKey```
+The HUD key generated by the client (will returns only for 'show' request).
+
+## Example
+
+### Request 
+#### state: show
```
Comment (Chasky-H) [inline actions/hud.md:70]: states should be PascalCase

File: actions/hud.md [hunk 13]
```diff
@@ -2,17 +2,18 @@
...
+In addition, the HUD can show a button that allow the user to close the HUD. the button will be shown only if ```CancelMessage``` text is provided.
+Once the Cancel Button is pressed, the client should close the hud and emit-event with ```CancelEventKey``` without data.
 
-the action have 3 states. 
-* show 
+the action has 3 states. 
+* Show 
     - the client should show the HUD on the screen for an ```Interval``` time
-* polling
+* Polling
```
Comment (Chasky-H) [inline actions/hud.md:14]: Poll

**Review by Chasky-H — CHANGES_REQUESTED** [review]

**Review by Chasky-H — APPROVED** [review]

**Pull request merged**